### PR TITLE
Handle workflow not found in replication

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Handle workflow not found in replication

**Why?**
Handle workflow not found in replication if the workflow passes retention and gets deleted in source cluster. We ignore this workflow and clean it up in target cluster.

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/history/replicationTaskExecutor.go [hunk 1]
```diff
@@ -153,17 +160,22 @@ func (e *replicationTaskExecutorImpl) handleActivityTask(
...
 			retryErr.StartEventVersion,
 			retryErr.EndEventId,
 			retryErr.EndEventVersion,
-		); resendErr != nil {
-			e.logger.Error("error resend history for sync activity", tag.Error(resendErr))
-			// should return the replication error, not the resending error
+		)
+		switch resendErr.(type) {
+		case *serviceerror.NotFound:
+			return e.cleanupWorkflowExecution(ctx, retryErr.NamespaceId, retryErr.WorkflowId, retryErr.RunId)
```
Comment (yiminc) [inline service/history/replicationTaskExecutor.go:174]: add comment

File: service/history/replicationTaskExecutor.go [hunk 2]
```diff
@@ -211,17 +223,22 @@ func (e *replicationTaskExecutorImpl) handleHistoryReplicationTask(
...
 			retryErr.RunId,
 			retryErr.StartEventId,
 			retryErr.StartEventVersion,
 			retryErr.EndEventId,
 			retryErr.EndEventVersion,
-		); resendErr != nil {
+		)
+		switch resendErr.(type) {
+		case *serviceerror.NotFound:
+			return e.cleanupWorkflowExecution(ctx, retryErr.NamespaceId, retryErr.WorkflowId, retryErr.RunId)
```
Comment (yiminc) [inline service/history/replicationTaskExecutor.go:237]: add comment

**Review by yiminc — APPROVED** [review]

File: service/history/replicationTaskExecutor.go [hunk 3]
```diff
@@ -256,3 +275,33 @@ FilterLoop:
...
+	defer func() { releaseFn(retErr) }()
+	mutableState, err := wfCtx.LoadWorkflowExecution(ctx)
+	if err != nil {
+		return err
+	}
+	lastWriteVersion, err := mutableState.GetLastWriteVersion()
+	if err != nil {
+		return err
+	}
+	return e.deleteManager.DeleteWorkflowExecution(
```
Comment (yycptt) [inline service/history/replicationTaskExecutor.go:298]: Synced offline. The implementation of `DeleteWorkflowExecution` will get workflow close event (for deleting vis record), which does not exist for running workflows. 

Need some refactoring work around deleting visibility records to make this work. Blocking this PR for now.

**Review by yycptt — CHANGES_REQUESTED** [review]

File: service/history/replicationTaskExecutor.go (hunk 3 above)
Comment (yux0) [reply service/history/replicationTaskExecutor.go:298]: Pending https://github.com/temporalio/temporal/pull/2636

**Review by yycptt — APPROVED** [review]

File: service/history/replicationTaskExecutor.go (hunk 3 above)
Comment (yycptt) [reply service/history/replicationTaskExecutor.go:298]: 👍 LGTM once #2636 is landed.

**Review by yycptt — APPROVED** [review]

**Pull request merged**